compiler/fir/providers/src/org/jetbrains/kotlin/fir/scopes/impl/FirSelfTypeScope.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.scopes.impl
+
+import org.jetbrains.kotlin.fir.declarations.FirMemberDeclaration
+import org.jetbrains.kotlin.fir.resolve.substitution.ConeSubstitutor
+import org.jetbrains.kotlin.fir.scopes.FirScope
+import org.jetbrains.kotlin.fir.symbols.impl.FirClassifierSymbol
+import org.jetbrains.kotlin.name.Name
+
+class FirSelfTypeScope(val memberDeclaration: FirMemberDeclaration) : FirScope() {
+
+    override fun processClassifiersByNameWithSubstitution(
+        name: Name,
+        processor: (FirClassifierSymbol<*>, ConeSubstitutor) -> Unit
+    ) {
+        if (name.asString() == "Self") {
+            val selfSymbol = memberDeclaration.typeParameters.find { it.symbol.name == Name.special("<Self>") }?.symbol
```

Comment:
It's better to extract `Name.special("<Self>")` to a constant

Reply:
Moved `Name.special("<Self>")` to  [SpecialNames](https://github.com/maxim092001/kotlin/blob/52535387eb5854a489835b8494a079d8a04b64f4/core/compiler.common/src/org/jetbrains/kotlin/name/SpecialNames.kt#L65)